src/css-loader-core/loader.js

```diff
@@ -1,112 +1,32 @@
-// Copied from https://github.com/css-modules/css-modules-loader-core
-
-import postcss from "postcss";
 import fs from "fs";
 import path from "path";
 
-import Parser from "./parser";
-
-class Core {
-  constructor(plugins) {
-    this.plugins = plugins || Core.defaultPlugins;
+export function resolveRelativeImport(importee, importer, projectRoot) {
+  const importerDir = path.dirname(importer);
+  let pathFromProjectRoot = path.resolve(projectRoot, importerDir, importee);
+
+  // if the path is not relative or absolute, try to resolve it in node_modules
+  if (importee[0] !== "." && importee[0] !== "/") {
+    try {
+      pathFromProjectRoot = require.resolve(importee);
+    } catch (e) {
+      // noop
+    }
   }
 
-  load(sourceString, sourcePath, trace, pathFetcher) {
-    let parser = new Parser(pathFetcher, trace);
-
-    return postcss(this.plugins.concat([parser.plugin()]))
-      .process(sourceString, { from: "/" + sourcePath })
-      .then((result) => {
-        return {
-          injectableSource: result.css,
-          exportTokens: parser.exportTokens,
-        };
-      });
-  }
+  return pathFromProjectRoot;
 }
 
-// Sorts dependencies in the following way:
-// AAA comes before AA and A
-// AB comes after AA and before A
-// All Bs come after all As
-// This ensures that the files are always returned in the following order:
-// - In the order they were required, except
-// - After all their dependencies
-const traceKeySorter = (a, b) => {
-  if (a.length < b.length) {
-    return a < b.substring(0, a.length) ? -1 : 1;
-  } else if (a.length > b.length) {
-    return a.substring(0, b.length) <= b ? -1 : 1;
-  } else {
-    return a < b ? -1 : 1;
-  }
-};
-
-export default class FileSystemLoader {
-  constructor(root, plugins) {
-    this.root = root;
-    this.sources = {};
-    this.traces = {};
-    this.importNr = 0;
-    this.core = new Core(plugins);
-    this.tokensByFile = {};
-  }
-
-  fetch(_newPath, relativeTo, _trace) {
-    let newPath = _newPath.replace(/^["']|["']$/g, ""),
-      trace = _trace || String.fromCharCode(this.importNr++);
+export default {
+  resolveId: (importee, importer, projectRoot) => {
+    return resolveRelativeImport(importee, importer, projectRoot)
+  },
+  resolve: (importee) => {
     return new Promise((resolve, reject) => {
-      let relativeDir = path.dirname(relativeTo),
-        rootRelativePath = path.resolve(relativeDir, newPath),
-        fileRelativePath = path.resolve(
-          path.join(this.root, relativeDir),
-          newPath
-        );
```

Comment:
@madyankin 
Btw, do you have a bit of insight on the old code? I'm not quite sure about the distinction between `rootRelativePath` and `fileRelativePath` here. They are equal as long as `this.root` is unset and I don't see why they need to be be different in the alternative case... so while refactoring I got rid of the distinction.
But since I don't completely trust the test suite anymore, I'm not sure if that's a correct step to take. Any idea?

Reply:
I believe I had the insight a couple of years ago or so, but not quite sure now. 

AFAIR, this is useful when you have CSS modules placed in some other location and want to resolve them properly, e.g., in Rails or Django frameworks. 

These files were in the `css-modules-loader-core` package without tests. Given I copied them here as a hotfix, I hadn't enough time to cover them with.